osu.Game/Screens/Play/KeyCounterCollection.cs

```diff
@@ -21,6 +23,8 @@ public class KeyCounterCollection : FillFlowContainer<KeyCounter>
         public readonly Bindable<bool> Visible = new Bindable<bool>(true);
         private readonly Bindable<bool> configVisibility = new Bindable<bool>();
 
+        private readonly Dictionary<string, List<KeyCounterMemento>> keyCountersState = new Dictionary<string, List<KeyCounterMemento>>();
```

Comment:
can't exist (ugly and should not be `string` based). why not store states inside the `KeyCounter` removing the need for this?